graphql-server/pkg/resolvers/validator.go

```diff
@@ -214,6 +215,60 @@ func (r *validatorResolver) Uptime(ctx context.Context, obj *models.Validator) (
 	return validator.SigningInfo.Uptime, nil
 }
 
+func (r *validatorResolver) ParticipatedProposalCount(ctx context.Context, obj *models.Validator) (int, error) {
+	validator, err := pkgContext.GetDataLoadersFromCtx(ctx).Validator.LoadValidatorWithInfoByConsensusAddress(obj.ConsensusAddress)
+	if err != nil {
+		return 0, err
+	}
+
+	if validator.Info == nil {
+		return 0, nil
+	}
+
+	return len(validator.Info.ProposalVotes), nil
+}
+
+func (r *validatorResolver) RelativeTotalProposalCount(ctx context.Context, obj *models.Validator) (int, error) {
+	validator, err := pkgContext.GetDataLoadersFromCtx(ctx).Validator.LoadValidatorWithInfoByConsensusAddress(obj.ConsensusAddress)
+	if err != nil {
+		return 0, err
+	}
+
+	if validator.SigningInfo == nil {
+		return 0, nil
+	}
+
+	proposals, err := pkgContext.GetDataLoadersFromCtx(ctx).Proposal.LoadAllProposals()
+	if err != nil {
+		return 0, err
+	}
+
+	var submitTime time.Time
+	if validator.SigningInfo.StartHeight != 0 {
+		startBlock, err := pkgContext.GetDataLoadersFromCtx(ctx).Block.LoadBlockByHeight(validator.SigningInfo.StartHeight)
+		if err != nil {
+			return 0, err
+		}
+
+		// Desynced
+		if startBlock == nil {
+			return 0, nil
+		}
+
+		submitTime = startBlock.Timestamp
+	}
+
+	totalCount := 0
+	// Count proposals that are submitted after the validator signing started
+	for _, proposal := range proposals {
+		if proposal.SubmitTime.After(submitTime) {
```

Comment:
make a query for this?

Reply:
bad to query all proposals & handle counting in golang layer

Reply:
oh I actually did query before but then thought it was a bit chur to call `SELECT * FROM proposal WHERE submitTime > startingTime` for each validator

Reply:
can you make a query for counting proposals since height, then make a dataloader?

counting is probably fine if indexing is good; loading all proposals from db to memory for each validator is probably more expensive